### PR TITLE
Don’t disable Server URL for anonymous logins

### PR DESCRIPTION
**Pull request by da2x**
Only disable the username and password fields.

### CODE REVIEW

File: src/options/views/tab-settings/vm-sync.vue [hunk 1]
```diff
@@ -22,9 +22,9 @@
       <label class="sync-server-url">
         <span v-text="i18n('labelSyncServerUrl')"></span>
         <input
-          type="text"
+          type="url"
```
Comment (tophf) [inline src/options/views/tab-settings/vm-sync.vue:25]: You also need to add `input[type=url]` in our [global css](https://github.com/violentmonkey/violentmonkey/blob/master/src/common/ui/style/style.css) in three places where `input[type=text]` is currently used.

**Review by tophf — CHANGES_REQUESTED** [review]

File: src/options/views/tab-settings/vm-sync.vue (hunk 1 above)
Comment (da2x) [reply src/options/views/tab-settings/vm-sync.vue:25]: Done.

**Review by tophf — APPROVED** [review]

**Comment by tophf** [comment]
Regarding `type=url`: actually, I'm not sure we want this. The documentation says it performs validation, which can be easily seen if we add something like `input[type=url]:invalid { border-color: red; }` but the input still allows nonsensical things like `foo: bar`.

**Comment by da2x** [comment]
Benefits of using the url input type:

* Browsers auto-complete knows to suggest URLs (not sure if any still suggest from browsing-history, but they at least suggest the right stuff from input-history). Specifically, this means you get the same partial auto-fill you get in the address field. Example: users can start typing "mailbo" and browsers will suggest https://webdav.mailbox.org/etc" if that’s in their history.
* Implies inputmode=url (touch-screen keyboards adds buttons for choosing TLDs and includes slashes, dots, and colons.)
* Implies spellcheck=false.

**Pull request merged**